src/script/interpreter.cpp

```diff
@@ -1153,7 +1177,10 @@ PrecomputedTransactionData::PrecomputedTransactionData(const CTransaction& txTo)
         ss << txTo;
         preTx.reset(zcash_transaction_precomputed_init(
             reinterpret_cast<const unsigned char*>(ss.data()),
-            ss.size()));
+            ss.size(), allPrevOutputs, allPrevOutputsLen));
+        if (preTx == nullptr) {
+            throw std::ios_base::failure("Invalid arguments to PrecomputedTransactionData");
+        }
```

Comment:
The way that I've handled this sort of thing elsewhere is to add a static factory method that does the initialization and returns in `std::optional`, and keep any fallible constructors private.